src/script/api/script_road.hpp

```diff
@@ -85,6 +85,42 @@ class ScriptRoad : public ScriptObject {
 		BT_TRUCK_STOP, ///< Build a truck stop
 	};
 
+	/**
+	 * A bitmap of all the possible road tiles and combinations.
+	 */
+	enum RoadTiles {
+		/* Note: these values represent part of the in-game RoadBits enum and shorthands to T-junctions */
+		ROADTILES_NONE = 0,                            ///< No road tiles
+		ROADTILES_NW = ::ROAD_NW,                      ///< North-west part
+		ROADTILES_SW = ::ROAD_SW,                      ///< South-west part
+		ROADTILES_SE = ::ROAD_SE,                      ///< South-east part
+		ROADTILES_NE = ::ROAD_NE,                      ///< North-east part
+		ROADTILES_X = ::ROAD_X,                        ///< Full road along the x-axis (south-west + north-east)
+		ROADTILES_Y = ::ROAD_Y,                        ///< Full road along the y-axis (north-west + south-east)
+		ROADTILES_N = ::ROAD_N,                        ///< Road at the two northern edges (corner, north-west + north-east)
+		ROADTILES_E = ::ROAD_E,                        ///< Road at the two eastern edges (corner, north-east + south-east)
+		ROADTILES_S = ::ROAD_S,                        ///< Road at the two southern edges (corner, south-east + south-west)
+		ROADTILES_W = ::ROAD_W,                        ///< Road at the two western edges (corner, south-west + north-west)
+		ROADTILES_S_NW = ROADTILES_S | ROADTILES_NW,   ///< T-junction, southern edges + north-west
+		ROADTILES_W_NE = ROADTILES_W | ROADTILES_NE,   ///< T-junction, western edges + north-east
+		ROADTILES_N_SE = ROADTILES_N | ROADTILES_SE,   ///< T-junction, northern edges + south-east
+		ROADTILES_E_SW = ROADTILES_E | ROADTILES_SW,   ///< T-junction, eastern edges + south-west
+		ROADTILES_ALL = ::ROAD_ALL,                    ///< Full 4-way crossing
+	};
+
+	/**
+	 * One-way info of the tile.
+	 */
+	enum OneWayInfo {
+		/* Note: Follows the same pattern as RoadCachedOneWayState. */
+		ONEWAY_NONE = 0,							///< Not a one-way road.
+		ONEWAY_NON_JUNCTION_A,				///< One-way road, no junctions, south-west to north-east + north-east to south-west
+		ONEWAY_NON_JUNCTION_B,				///< One-way road, no junctions, north-west to south-east + south-east to north-west
+		ONEWAY_NO_ACCESS,							///< Road is disallowed in both directions
+		ONEWAY_SIDE_JUNCTION,					///< One-way road junction, can exit
+		ONEWAY_SIDE_JUNCTION_NO_EXIT,	///< One-way road junction, can't exit
+	};
```

Comment:
RoadCachedOneWayState is an internal structure which is subject to change.
It'd probably be better to use DisallowedRoadDirections/GetDisallowedRoadDirections.